### PR TITLE
Add precommit and isort

### PR DESCRIPTION
**Pull request by patrick91**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+repos:
+  - repo: git://github.com/pre-commit/pre-commit-hooks
+    rev: v1.3.0
```
Comment (jkimbo) [inline .pre-commit-config.yaml]:
```suggestion
    rev: v2.1.0
```

The current version on the graphene repo is v2.1.0

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      - id: end-of-file-fixer
+        exclude: ^docs/.*$
+      - id: trailing-whitespace
+        exclude: README.md
+      - id: pretty-format-json
+        args:
+          - --autofix
+      - id: flake8
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v1.4.0
```
Comment (jkimbo) [inline .pre-commit-config.yaml]:
```suggestion
    rev: v1.12.0
```

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    hooks:
+      - id: pyupgrade
+  - repo: https://github.com/ambv/black
+    rev: 18.6b4
```
Comment (jkimbo) [inline .pre-commit-config.yaml]:
```suggestion
    rev: 18.9b0
```

Also has black been run across the code base?

File: .isort.cfg [hunk 4]
```diff
@@ -0,0 +1,2 @@
+[settings]
+known_third_party = aniso8601,graphql,graphql_relay,promise,pytest,pytz,pyutils,setuptools,six,snapshottest,sphinx_graphene_theme
```
Comment (danpalmer) [inline .isort.cfg]: Might be nice to put these on separate lines for easier readability and less diff noise in the future.

**Comment by dulmandakh** [comment]
LGTM. hey @patrick91, could you please make changes suggested by @danpalmer and @jkimbo.

**Comment by jkimbo** [comment]
@patrick91 are you still interested in maintaining this PR? I've also recently created a PR (https://github.com/graphql-python/graphene-django/pull/667) that adds black and flake8 checks as part of the travis build (instead of using pre-commit) so this PR will have to be updated once that is merged.

**Pull request closed without merging**